src/app/pages/users/user/payment-methods/payment-method/stripe/stripe-payment-method.component.ts

```diff
@@ -143,72 +144,61 @@ export class StripePaymentMethodComponent implements OnInit {
-      // Step #1 - Confirm the SetupIntent with data provided and carry out 3DS
-      // c.f. https://stripe.com/docs/js/setup_intents/confirm_card_setup
-      // -----------------------------------------------------------------------------------------------
-      const setupIntent: any = response?.internalData;
-      // TODO: handle spinner .hide / .show in a better way ? to be tested in prod // we cannot anymore reclick save button twice
-      // setTimeout doesn't work as expected - it never hides...
-      // setTimeout(function() {
-      this.spinnerService.hide();
-      // }, 4000);
-      // eslint-disable-next-line max-len
-      const result: { setupIntent?: SetupIntent; error?: StripeError } = await this.getStripeFacade().confirmCardSetup( setupIntent.client_secret, {
-        payment_method: {
-          card: this.cardNumber
-          // TODO: put email and address
-          // billing_details: {
-          //   name: this.centralServerService.getCurrentUserSubject().value.email + new Date(),
-          // },
-        },
-      });
-      this.spinnerService.show();
-      if (result.error) {
-        operationResult = result;
-      } else {
-        operationResult = this.attachPaymentMethod(result);
-      }
-    } catch (error) {
-      Utils.handleHttpError(error, this.router, this.messageService, this.centralServerService, 'general.unexpected_error_backend');
+      return response?.internalData;
     } finally {
       this.spinnerService.hide();
     }
-    return operationResult;
   }
-  // -----------------------------------------------------------------------------------------------
-  // Step #2 - Really attach the payment method / not called when 3DS failed
-  // -----------------------------------------------------------------------------------------------
-  private async attachPaymentMethod(operationResult: {setupIntent?: SetupIntent; error?: StripeError}) {
-    const response: BillingOperationResult = await this.centralServerService.setupPaymentMethod({
-      setupIntentId: operationResult.setupIntent?.id,
-      paymentMethodId: operationResult.setupIntent?.payment_method,
-      userID: this.userID
-    }).toPromise();
-    return response;
+
+  private async confirmSetupIntent(setupIntent: any): Promise<{ setupIntent?: SetupIntent; error?: StripeError }> {
+    const result: { setupIntent?: SetupIntent; error?: StripeError } = await this.getStripeFacade().confirmCardSetup( setupIntent.client_secret, {
+      payment_method: {
+        card: this.cardNumber
+      },
+    });
+    return result;
   }
 
-  public close(saved: boolean = false) {
-    if (this.inDialog) {
-      this.dialogRef.close(saved);
+  private async attachPaymentMethod(operationResult: {setupIntent?: SetupIntent; error?: StripeError}) {
+    try {
+      this.spinnerService.show();
+      const response: BillingOperationResult = await this.centralServerService.setupPaymentMethod({
```

Comment:
Interesting technic to not call subscribe() and use async/await.
Should we generalize it?

Reply:
@ClaudeROSSI Any news?
In Angular in general we use subscribe (RxJs) not promises.

Reply:
> > Why we need a Promise here!? Is it a constraint from Stripe API?
> 
> This is just to make the code human-readable for the caller! async / await are (most of the time) just syntactic sugar

Okay, then as it'll be deprecated, use RxJs ;-)